.buildkite/pipeline.yml

```diff
@@ -344,6 +344,25 @@ steps:
       commit: "${BUILDKITE_COMMIT}"
       branch: "${BUILDKITE_BRANCH}"
 
+  # Trigger for pull requests - Serverless Beats Tests
+  # this should help detecting issues earlier in the development cycle
+  - label: "Trigger Serverless Beats Tests"
+    if: build.pull_request.id != null
+    plugins:
+      - monorepo-diff#v1.2.0:
+          diff: "git diff --name-only origin/${GITHUB_PR_TARGET_BRANCH}...HEAD"
+          interpolation: false
+          watch:
+            - path:
+                - .buildkite/serverless.beats.tests.yml
+                - .buildkite/scripts/steps/beats_tests.sh
+                - .buildkite/hooks/pre-command
+              config:
+                trigger: "elastic-agent-extended-testing"
```

Comment:
I use `trigger`; otherwise, the if condition in the `pre-command` will require granting access to the main pipeline or include more sophisticated conditionals.

The trigger step will make the GitHub build comment not work nicely for this step; however, for simplicity, I think it's worth keeping it simple for now, even if the github build comment won't contain those details clearly - this pipeline is not intended to change often.

I decided not to use the BK PR bot, because it does not support skipping the very first build for a PR, see https://github.com/elastic/logstash/pull/18327 (that introduced an artificial pipeline to manage those cases)